\section{CRISPR-Cas Systems}
\subsection{What Are They?}
\ac{crspc} systems are sets of nucleotide motifs (spacers) interspaced with nucleotide repeats (\ac{crsp}s) and \ac{cas} proteins (usually adjacent to the \ac{crsp} motifs) that have an adaptive immune function in many bacteria and archaea \citep{crispgen}.
Each nucleotide motif is indicative of some DNA sequence that was taken up previously by the host and serves as a marker for the \ac{cas} proteins to degrade any foreign DNA matching this motif \citep{crispgen}.
If a bacterium which possesses a \ac{crspc} system is infected with a phage and survives, a motif representative of that phage can be integrated into the \ac{crsp} repeats so that when the, bacterium is reinfected with the same phage strain it will be detected and degraded by a \ac{cas} protein before genomic integration can occur.
\ac{crsp} is an adaptive immune system, as the bacterium acquire resistance after an unsuccessful infection through spacer integration.\par
Although \ac{crspc} is primarily considered an immune system, non-viral spacers representative of bacterial \ac{mge}s have been found to compose the majority of (detectable) \ac{crsp} spacers \citep{nonvspacer}.
In fact, many spacers have no detectable match to a viral sequence, being termed \ac{crsp} ``dark matter'', indicating that knowledge about the acquisition of spacers and their effects on bacterial gene dynamics leaves much to be desired \citep{nonvspacer}.
\subsection{Diversity, Ubiquity And Detection}
As of 2017, over $45\%$ of bacterial genomes analyzed ($n=6782$) appear to contain \ac{crsp} motifs \citep{crispdb}.
Moreover, \ac{crsp} motifs show significant diversity between organisms, since they represent a chronological history of viral infection or \ac{mge} ``infection'' for that specific organism \citep{crispgen}.
\ac{cas} proteins themselves show significant diversification, segregating into entirely different \ac{crspc} systems \citep{evocas}.
There still exist many bacterial strains, and even entire genera with no \textit{known} \ac{crspc} systems, although they may have simply not been discovered yet \citep{ineqcas,casguild}.\par
Between $11\%-28\%$ of sequenced genomes have either only \ac{crsp} repeats \textit{or} \ac{cas} loci, but not both \citep{ineqcas}.
There also exist repetitive motifs that may superficially resemble \ac{crsp}s, but have low spacer diversity and no \ac{cas} genes \citep{ineqcas}.
False detection of \ac{crsp} systems is significantly increased by only examining repeat-spacer structural patterns, other parameters such as spacer dissimilarity and genomic context should be considered to reduce false positives \citep{ineqcas}.
Especially as sequencing efforts continue, better mechanistic understandings of \ac{crsp} systems develop and \ac{crsp} systems themselves propagate and transfer between bacteria they will continue to become more relevant and diverse \citep{crispgen}.
Furthermore, the diversification of CRISPR-Cas systems is driven further by \ac{hgt} acting on \ac{crsp} and \ac{cas} components independently, adding another level of complexity to the propagation of CRISPR-Cas systems \citep{crispgen}.
\subsection{Applications In Biotechnology}
While \ac{crsp}s are interesting systems to study from a microbiological perspective, much of the current research interest (and funding) is motivated by applications of \ac{crsp} to gene editing.
The CRISPR-Cas9 system has been adapted into a simple, efficient tool for gene editing in both prokaryotes and eukaryotes \citep{crispgen}.
The \ac{cas}-9 protein induces a double-strand break to a region homologous to a guide RNA, which can be synthesized by a researcher.
The break will then be re-annealed by DNA repair enzymes, often introducing errors (insertions,deletions,etc.) into the sequence, disrupting gene function \citep{crispgen}.
A gene can also be inserted at the break point via homology directed repair by including DNA sequence with flanking arms homologous to the break region \citep{crispgen}.
