\section{Future Work}
There are multiple ways to expand this analysis to answer other questions related to the transfer of genes.
As \ac{hgt} inference methods improve and it becomes possible to discern the direction of transfer with confidence, a whole new set of techniques become available for study.
Some possible ways to extend this analysis are:
\begin{itemize}
    \item \textbf{Inferring direction}: Directed networks have a host of available analytic tools undirected networks do not
    \item \textbf{Gene function analysis}: Considering the transfer dynamics of different functional classes of genes
    \item \textbf{Studying movement of \ac{crsp} systems}: Studying how frequently \ac{crsp} systems themselves are transferred from arrays, \textit{Cas} genes
    \item \textbf{Intergenic comparisons}: Combine any set of fasta files from \ac{otu}s for analyzing transfer dynamics
    \item \textbf{Continuous \ac{crsp} activity}: Labelling nodes by estimated \ac{crsp} activity (array length, transciptomic data, etc.)
    \item \textbf{Considering bacterial ecology and environments}: Consider geographically close \ac{otu}s or differences between networks due to environmental factors
\end{itemize}
Ultimately, this pipeline provides a fairly straightforward way to study trends in \ac{hgt} for a set of bacterial \ac{otu}s.
%If one is interested in studying the transfer patterns of a specific grouping of genes, either by function, common structural motifs, sequence composition, expression pattern this type of analysis is highly suitable.
%For example, the transfer of CRISPR systems themselves is something largely unstudied.
%Networks can be constructed from Cas and Cfp1 genes as well as identified \ac{crsp} arrays to estimate how often \ac{crspc} systems themselves move around communities.
%Networks constructed from ribosomal genes can be used as a reference point for what very little transfer looks like.
