\section{CRISPR-Cas Systems}
\subsection{What Are They?}
\ac{crspc} systems are sets of nucleotide motifs (spacers) interspaced with nucleotide repeats (\ac{crsp}s) and \ac{cas} proteins (usually adjacent to the \ac{crsp} motifs) that serve an adaptive immune function in many bacteria and archaea\citep{crispgen}.
Each nucleotide motif is indicative of some DNA sequence that was taken up previously by the host and serves as a marker for the \ac{cas} proteins to degrade any foreign DNA matching this motif\citep{crispgen}.
For example, if a bacterium is infected with a phage and survives, a motif representative of that phage can be integrated into the \ac{crsp} repeats so that when the phage attempts infection again, it will be degraded by a \ac{cas} protein before genomic integration can occur.
\ac{crsp} is an adaptive immune systems, as it adapts resistance after an initial infection, which is generally required for the spacer sequence to be integrated.\par
Despite seeming to function primarily as an immune system, non-viral spacers representative of bacterial \ac{mge}s have been found to compose a majority of (detectable) \ac{crsp} spacers\citep{nonvspacer}.
In fact, many spacers were found to have no detectable match, being termed \ac{crsp} "dark matter", indicating that knowledge about the acquisition of spacers and their effects on bacterial gene dynamics leaves much to be desired\citep{nonvspacer}.
\subsection{Diversity, Ubiquity And Detection}
As of 2017, over $45\%$ of bacterial genomes analyzed ($n=6782$) have been found to have \ac{crsp} motifs\citep{crispdb}.
\ac{crsp} motifs show significant diversity between organisms, as expected since they represent a chronological history of viral infection or \ac{mge} "infection"\citep{crispgen}.
But even \ac{cas} proteins themselves also show diversity and evolution, segregating into entirely different \ac{crspc} systems\citep{evocas}.
There still exist many bacterial strains, and even entire genera with no known \ac{crspc} systems, although the emphasis should be placed on the word \textit{known}\citep{ineqcas,casguild}.\par
Between $11\%-28\%$ of sequenced genomes have either only \ac{crsp} repeats \textit{or} \ac{cas} loci, but not both\citep{ineqcas}.
There also exist repeat motifs that may superficially resemble \ac{crsp}s, but have low spacer diversity and no \ac{cas} genes\citep{ineqcas}.
False detection of \ac{crsp} systems is significantly increased by only considering repeat-spacer structural patterns, and considerations of more information, such as spacer dissimilarity and genomic context are necessary for reducing false positives\citep{ineqcas}.
Especially as sequencing efforts continue, better mechanistic understandings of \ac{crsp} systems develop and \ac{crsp} systems themselves propagate and transfer between bacteria they will continue to become more relevant and diverse\citep{crispgen}.
\subsection{Applications In Biotechnology}
While \ac{crsp}s are interesting systems to study from a microbiological perspective, much of the current research interest (and funding) is motivated by applications of \ac{crsp} to gene editing.
CRISPR-Cas9 has been developed into a simple, efficient tool for gene editing in both prokaryotes and eukaryotes\citep{crispgen}.
Using the \ac{cas}-9 protein which induces a double-strand break to a region homologous to a specified guide RNA.
The break will then  re-anneal (likely) incorrectly, removing gene function\citep{crispgen}.
A gene can also be inserted at the break point via homology directed repair by including DNA sequence with flanking arms homologous to the break region\citep{crispgen}.

